src/cpp/session/SessionMain.cpp

```diff
@@ -1642,6 +1643,12 @@ Error runPreflightScript()
    // always return success
    return Success();
 }
+
+void exitEarly(int status)
+{
+   FileLock::cleanUp();
+   ::exit(status);
```

Comment:
I agree that it's not a great idea to shadow ::exit



On Thu, Nov 12, 2015 at 12:58 PM, Kevin Ushey <notifications@github.com>
wrote:

> In src/cpp/session/SessionMain.cpp
> <https://github.com/rstudio/rstudio/pull/526#discussion_r44690555>:
>
> > @@ -1642,6 +1643,12 @@ Error runPreflightScript()
> >     // always return success
> >     return Success();
> >  }
> > +
> > +void exitEarly(int status)
> > +{
> > +   FileLock::cleanUp();
> > +   ::exit(status);
>
> You're right, but shadowing exit seems a little dicey to me (what if two
> competing headers want to shadow exit? Unlikely now but could happen
> somewhere down the road...)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/526/files#r44690555>.
>